### ISSUE TITLE
LMMS freezes at 100% on export 

### ISSUE THREAD

**Issue by ViperZer0** [LMMS/lmms#5664]
### Bug Summary

LMMS freezes at 100% on any exports that I try. I have tried to export three different projects, one which was new, and two older projects which I know had at one point successfully exported. Each one freezes at 100% and one closes entirely. Running LMMS from a terminal gives me the following error when the freeze/crash happens: 
`QMetaMethod::invoke: Unable to handle unregistered datatype 'MidiTime'`

The project that crashes gives an additional segfault.

#### Steps to reproduce

- Create any project, the behavior does not seem to be affected at all by the contents of the project.

#### Expected behavior

Project should export successfully.

#### Actual behavior

LMMS freezes and sometimes segfaults at 100% exported, throwing the error
`QMetaMethod::invoke: Unable to handle unregistered datatype 'MidiTime'`

#### Affected LMMS versions
Version 1.2.2
Arch Linux
JACK Audio

**Comment by trebmuh** [comment]
I've tried with the demo "CapDan-ReggaeTry".
I've got the same kind of behavior where LMMS doesn't crash, but it freezes.
Launched from a terminal, it says the same:
`QMetaMethod::invoke: Unable to handle unregistered datatype 'MidiTime'`

LMMS 1.2.2, JACK, options:
![export-options](https://user-images.githubusercontent.com/8705846/91973132-ff384880-ed1b-11ea-909b-6a251d633dde.png)

Note that even if LMMS freezes, the WAV file seems to be correctly exported (ie: I can read it with VLC).

**Comment by Spekular** [comment]
How long are you waiting to determine that LMMS is frozen? I know on my system, LMMS sometimes sits several seconds at 100% before the export completes.

You mention that you've succesfully exported projects before; what's changed since then? What was the latest configuration that worked?

**Comment by trebmuh** [comment]
@Spekular I've just run a test for 5 full minutes and it still is frozen.

**Comment by Daniele71** [comment]
It happens to me too. opensuse TW

**Comment by musikBear** [comment]
@ViperZer0 & @Daniele71 
Are you **both** on Arch??
We need more information from you @Daniele71 !
I would like you both to try and change the backend to SDL, and retry to export, that way we can point the finger at *JACK Audio*
Change of backend:
https://lmms.io/wiki/index.php?title=LMMS_Settings#Audio_Settings
If you *still* freeze we can point at Arch

**Comment by Daniele71** [comment]
> @ViperZer0 & @Daniele71
> Are you **both** on Arch??
> We need more information from you @Daniele71 !
> I would like you both to try and change the backend to SDL, and retry to export, that way we can point the finger at _JACK Audio_
> Change of backend:
> https://lmms.io/wiki/index.php?title=LMMS_Settings#Audio_Settings
> If you _still_ freeze we can point at Arch

Opensuse Tumbleweed here. It works with SDL backend !

**Comment by trebmuh** [comment]
Same for me. Tested the export with SDL, then tested with ALSA and it works fine.

**Comment by musikBear** [comment]
Super! Thank you *both* @trebmuh & @Daniele71 for your feedback! 
This issue is something related to JACK, or a *version* of JACK.
@trebmuh & @Daniele71
Afaik SDL output is not inferior to JACK, so you should use SDL for now :)

**Comment by trebmuh** [comment]
You're welcome. If that can help, I'm using a debian 1.9.14~dfsg-1 backport version of JACK2.

**Comment by firewall1110** [comment]
The same problem is in master (I downloaded&compiled it at 15.09.2020). 
Only after export and before freezing message **is not**:
_QMetaMethod::invoke: Unable to handle unregistered datatype 'MidiTime'_
**but** (something about this):
_PERFLOG |                Project Render  | 35.46user, 1.40system 17.39elapsed_

I use Debian Buster : _libjack-jack2-dev 1.9.12~dfsg-2_ 

P.S.
Terminal output:
_QMetaMethod::invoke: Unable to handle unregistered datatype 'MidiTime'_
is always after rendering - not only in Jack audio device, and after this  only with Jack aufio device LMMS freezing. .

**Comment by firewall1110** [comment]
The same problem is in master (I downloaded&compiled it at 15.09.2020). 
Only after export and before freezing message **is not**:
_QMetaMethod::invoke: Unable to handle unregistered datatype 'MidiTime'_
**but** (something about this):
_PERFLOG |                Project Render  | 35.46user, 1.40system 17.39elapsed_

I use Debian Buster : _libjack-jack2-dev 1.9.12~dfsg-2_ 

P.S.
Terminal output:
_QMetaMethod::invoke: Unable to handle unregistered datatype 'MidiTime'_
is always after rendering - not only in Jack audio device, and after this  only with Jack aufio device LMMS freezing. .

**Comment by DomClark** [comment]
Can someone provide a backtrace of the hang? See https://github.com/LMMS/lmms/wiki/Debugging-LMMS.

**Comment by firewall1110** [comment]
Is it what you asked? 

P.S. 

It seems, that lmms freeze on deadlock :
- CPU load is 0;
- no exceptions;

**Comment by firewall1110** [comment]
Some problems to insert file, so :
Id   Target Id                                           Frame 
* 1    Thread 0x7ffff37aaec0 (LWP 17817) "lmms"            futex_wait_cancelable (private=0, expected=0, futex_word=0x555558e668f0)
    at ../sysdeps/unix/sysv/linux/futex-internal.h:88
[...]
  6    Thread 0x7fffe77ee700 (LWP 17837) "lmms"            __libc_read (
    nbytes=4, buf=0x7fffe77edc40, fd=15)
    at ../sysdeps/unix/sysv/linux/read.c:26
  7    Thread 0x7fffc7187700 (LWP 17838) "MidiAlsaSeq"     0x00007ffff5f58819 in __GI___poll (fds=0x7fffc0004e10, nfds=2, timeout=250)
    at ../sysdeps/unix/sysv/linux/poll.c:29
  9    Thread 0x7fffd04ba700 (LWP 17840) "lmms"            syscall ()
    at ../sysdeps/unix/sysv/linux/x86_64/syscall.S:38
...

#0  0x00007ffff7f8b00c in futex_wait_cancelable
    (private=0, expected=0, futex_word=0x555558e668f0)
    at ../sysdeps/unix/sysv/linux/futex-internal.h:88
[...]
#8  0x00005555556f1afe in RenderManager::~RenderManager()
    (this=0x5555567bece0, __in_chrg=<optimized out>)
    at /home/wb/build/lmms_build/test_xz_debuging/lmms/src/core/RenderManager.cpp:56
#9  0x0000555555741d15 in ExportProjectDialog::accept() (this=0x7fffffffcaf0)
    at /home/wb/build/lmms_build/test_xz_debuging/lmms/src/gui/ExportProjectDialog.cpp:114
#10 0x00005555557f089d in ExportProjectDialog::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)
    (_o=0x7fffffffcaf0, _c=QMetaObject::InvokeMetaMethod, _id=2, _a=0x7fffffffc420)
    at /home/wb/build/lmms_build/test_xz_debuging/lmms/build/src/lmmsobjs_autogen/DGKZTCOIDE/moc_ExportProjectDialog.cpp:56
#11 0x00007ffff729dc90 in QMetaObject::activate(QObject*, QMetaObject const*, int, void**) () at /lib/x86_64-linux-gnu/libQtCore.so.4
#12 0x00005555557f8d57 in RenderManager::finished() (this=0x5555567bece0)
    at /home/wb/build/lmms_build/test_xz_debuging/lmms/build/src/lmmsobjs_autogen/DGKZTCOIDE/moc_RenderManager.cpp:113
#13 0x00005555556f1bf9 in RenderManager::renderNextTrack()
    (this=0x5555567bece0)
    at /home/wb/build/lmms_build/test_xz_debuging/lmms/src/core/RenderManager.cpp:78
[...]

P.S.
- CPU load = 0
- no exceptions, faults, ... 
Deadlock?

**Comment by firewall1110** [comment]
Id   Target Id                                           Frame
* 1    Thread 0x7ffff37aaec0 (LWP 17817) "lmms"            futex_wait_cancelable (private=0, expected=0, futex_word=0x555558e668f0)
    at ../sysdeps/unix/sysv/linux/futex-internal.h:88
  2    Thread 0x7fffefeed700 (LWP 17825) "QThread"         futex_wait_cancelable (private=0, expected=0, futex_word=0x555555b34ed0)
    at ../sysdeps/unix/sysv/linux/futex-internal.h:88
  3    Thread 0x7fffe76ec700 (LWP 17826) "QThread"         futex_wait_cancelable (private=0, expected=0, futex_word=0x555555b34ed0)
    at ../sysdeps/unix/sysv/linux/futex-internal.h:88
  4    Thread 0x7fffef6ec700 (LWP 17827) "QThread"         futex_wait_cancelable (private=0, expected=0, futex_word=0x555555b34ed0)
    at ../sysdeps/unix/sysv/linux/futex-internal.h:88
  5    Thread 0x7fffec08d700 (LWP 17836) "lmms"            futex_wait_cancelable (private=0, expected=0, futex_word=0x555555cbd808)
    at ../sysdeps/unix/sysv/linux/futex-internal.h:88
  6    Thread 0x7fffe77ee700 (LWP 17837) "lmms"            __libc_read (
    nbytes=4, buf=0x7fffe77edc40, fd=15)
    at ../sysdeps/unix/sysv/linux/read.c:26
  7    Thread 0x7fffc7187700 (LWP 17838) "MidiAlsaSeq"     0x00007ffff5f58819 in __GI___poll (fds=0x7fffc0004e10, nfds=2, timeout=250)
    at ../sysdeps/unix/sysv/linux/poll.c:29
  9    Thread 0x7fffd04ba700 (LWP 17840) "lmms"            syscall ()
    at ../sysdeps/unix/sysv/linux/x86_64/syscall.S:38
  10   Thread 0x7fffc4edf700 (LWP 17917) "QFileInfoGather" futex_wait_cancelable (private=0, expected=0, futex_word=0x55555693cef4)
    at ../sysdeps/unix/sysv/linux/futex-internal.h:88
  11   Thread 0x7fffb7fff700 (LWP 17918) "QInotifyFileSys" 0x00007ffff5f58819 in __GI___poll (fds=0x7fffac0052f0, nfds=2, timeout=-1)
    at ../sysdeps/unix/sysv/linux/poll.c:29
  13   Thread 0x7fffc6986700 (LWP 17964) "QThread"         futex_wait_cancelable (private=0, expected=0, futex_word=0x555555aac150)
    at ../sysdeps/unix/sysv/linux/futex-internal.h:88
#0  0x00007ffff7f8b00c in futex_wait_cancelable
    (private=0, expected=0, futex_word=0x555558e668f0)
    at ../sysdeps/unix/sysv/linux/futex-internal.h:88
#1  0x00007ffff7f8b00c in __pthread_cond_wait_common
    (abstime=0x0, mutex=0x555558e668a0, cond=0x555558e668c8)
    at pthread_cond_wait.c:502
#2  0x00007ffff7f8b00c in __pthread_cond_wait
    (cond=0x555558e668c8, mutex=0x555558e668a0) at pthread_cond_wait.c:655
#3  0x00007ffff7190d23 in QWaitCondition::wait(QMutex*, unsigned long) ()
    at /lib/x86_64-linux-gnu/libQtCore.so.4
#4  0x00007ffff71902ae in QThread::wait(unsigned long) ()
    at /lib/x86_64-linux-gnu/libQtCore.so.4
#5  0x00005555556d319a in Mixer::stopProcessing() (this=0x555555a71cc0)
    at /home/wb/build/lmms_build/test_xz_debuging/lmms/src/core/Mixer.cpp:250
#6  0x00005555556d4172 in Mixer::changeQuality(Mixer::qualitySettings const&)
    (this=0x555555a71cc0, _qs=...)
    at /home/wb/build/lmms_build/test_xz_debuging/lmms/src/core/Mixer.cpp:564
#7  0x00005555556f1aa6 in RenderManager::~RenderManager()
    (this=0x5555567bece0, __in_chrg=<optimized out>)
    at /home/wb/build/lmms_build/test_xz_debuging/lmms/src/core/RenderManager.cpp:55
#8  0x00005555556f1afe in RenderManager::~RenderManager()
    (this=0x5555567bece0, __in_chrg=<optimized out>)
    at /home/wb/build/lmms_build/test_xz_debuging/lmms/src/core/RenderManager.cpp:56
#9  0x0000555555741d15 in ExportProjectDialog::accept() (this=0x7fffffffcaf0)
    at /home/wb/build/lmms_build/test_xz_debuging/lmms/src/gui/ExportProjectDialog.cpp:114
#10 0x00005555557f089d in ExportProjectDialog::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)
    (_o=0x7fffffffcaf0, _c=QMetaObject::InvokeMetaMethod, _id=2, _a=0x7fffffffc420)
    at /home/wb/build/lmms_build/test_xz_debuging/lmms/build/src/lmmsobjs_autogen/DGKZTCOIDE/moc_ExportProjectDialog.cpp:56
#11 0x00007ffff729dc90 in QMetaObject::activate(QObject*, QMetaObject const*, int, void**) () at /lib/x86_64-linux-gnu/libQtCore.so.4
#12 0x00005555557f8d57 in RenderManager::finished() (this=0x5555567bece0)
    at /home/wb/build/lmms_build/test_xz_debuging/lmms/build/src/lmmsobjs_autogen/DGKZTCOIDE/moc_RenderManager.cpp:113
#13 0x00005555556f1bf9 in RenderManager::renderNextTrack()
    (this=0x5555567bece0)
    at /home/wb/build/lmms_build/test_xz_debuging/lmms/src/core/RenderManager.cpp:78
#14 0x00005555557f8bd5 in RenderManager::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)
    (_o=0x5555567bece0, _c=QMetaObject::InvokeMetaMethod, _id=2, _a=0x7fffb80038e0)
    at /home/wb/build/lmms_build/test_xz_debuging/lmms/build/src/lmmsobjs_autogen/DGKZTCOIDE/moc_RenderManager.cpp:57
#15 0x00007ffff72a2601 in QObject::event(QEvent*) ()
    at /lib/x86_64-linux-gnu/libQtCore.so.4
#16 0x00007ffff7682a2c in QApplicationPrivate::notify_helper(QObject*, QEvent*)
    () at /lib/x86_64-linux-gnu/libQtGui.so.4
#17 0x00007ffff7689212 in QApplication::notify(QObject*, QEvent*) ()
    at /lib/x86_64-linux-gnu/libQtGui.so.4
#18 0x00007ffff7289afb in QCoreApplication::notifyInternal(QObject*, QEvent*)
    () at /lib/x86_64-linux-gnu/libQtCore.so.4
#19 0x00007ffff728c815 in QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) () at /lib/x86_64-linux-gnu/libQtCore.so.4
#20 0x00007ffff72b67b3 in  () at /lib/x86_64-linux-gnu/libQtCore.so.4
#21 0x00007ffff5b36f2e in g_main_context_dispatch ()
    at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#22 0x00007ffff5b371c8 in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#23 0x00007ffff5b3725c in g_main_context_iteration ()
    at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#24 0x00007ffff72b68e1 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () at /lib/x86_64-linux-gnu/libQtCore.so.4
#25 0x00007ffff77221f7 in  () at /lib/x86_64-linux-gnu/libQtGui.so.4
#26 0x00007ffff728838f in QEventLoop::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () at /lib/x86_64-linux-gnu/libQtCore.so.4
#27 0x00007ffff728865e in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () at /lib/x86_64-linux-gnu/libQtCore.so.4
#28 0x00007ffff7b1350f in QDialog::exec() ()
    at /lib/x86_64-linux-gnu/libQtGui.so.4
#29 0x00005555556ff2be in Song::exportProject(bool)
    (this=0x555555ad0b90, multiExport=false)
    at /home/wb/build/lmms_build/test_xz_debuging/lmms/src/core/Song.cpp:1436
#30 0x00005555557fa67c in Song::qt_static_metacall(QObject*, QMetaObject::Call, int, void**)
    (_o=0x555555ad0b90, _c=QMetaObject::InvokeMetaMethod, _id=19, _a=0x7fffffffcf30)
    at /home/wb/build/lmms_build/test_xz_debuging/lmms/build/src/lmmsobjs_autogen/DGKZTCOIDE/moc_Song.cpp:124
#31 0x00007ffff729dc90 in QMetaObject::activate(QObject*, QMetaObject const*, int, void**) () at /lib/x86_64-linux-gnu/libQtCore.so.4
#32 0x00007ffff767d012 in QAction::triggered(bool) ()
    at /lib/x86_64-linux-gnu/libQtGui.so.4
#33 0x00007ffff767e2a7 in QAction::activate(QAction::ActionEvent) ()
    at /lib/x86_64-linux-gnu/libQtGui.so.4
#34 0x00007ffff7a94684 in  () at /lib/x86_64-linux-gnu/libQtGui.so.4
#35 0x00007ffff7a98ad1 in  () at /lib/x86_64-linux-gnu/libQtGui.so.4
#36 0x00007ffff76d4a28 in QWidget::event(QEvent*) ()
    at /lib/x86_64-linux-gnu/libQtGui.so.4
#37 0x00007ffff7a9c52b in QMenu::event(QEvent*) ()
    at /lib/x86_64-linux-gnu/libQtGui.so.4
#38 0x00007ffff7682a2c in QApplicationPrivate::notify_helper(QObject*, QEvent*)
    () at /lib/x86_64-linux-gnu/libQtGui.so.4
#39 0x00007ffff76893fc in QApplication::notify(QObject*, QEvent*) ()
    at /lib/x86_64-linux-gnu/libQtGui.so.4
#40 0x00007ffff7289afb in QCoreApplication::notifyInternal(QObject*, QEvent*)
    () at /lib/x86_64-linux-gnu/libQtCore.so.4
#41 0x00007ffff7688b3b in QApplicationPrivate::sendMouseEvent(QWidget*, QMouseEvent*, QWidget*, QWidget*, QWidget**, QPointer<QWidget>&, bool) ()
    at /lib/x86_64-linux-gnu/libQtGui.so.4
#42 0x00007ffff76fc1d6 in  () at /lib/x86_64-linux-gnu/libQtGui.so.4
#43 0x00007ffff76fa9d0 in QApplication::x11ProcessEvent(_XEvent*) ()
    at /lib/x86_64-linux-gnu/libQtGui.so.4
#44 0x00007ffff772206c in  () at /lib/x86_64-linux-gnu/libQtGui.so.4
#45 0x00007ffff5b36f2e in g_main_context_dispatch ()
    at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#46 0x00007ffff5b371c8 in  () at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#47 0x00007ffff5b3725c in g_main_context_iteration ()
    at /lib/x86_64-linux-gnu/libglib-2.0.so.0
#48 0x00007ffff72b68e1 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () at /lib/x86_64-linux-gnu/libQtCore.so.4
#49 0x00007ffff77221f7 in  () at /lib/x86_64-linux-gnu/libQtGui.so.4
#50 0x00007ffff728838f in QEventLoop::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () at /lib/x86_64-linux-gnu/libQtCore.so.4
#51 0x00007ffff728865e in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () at /lib/x86_64-linux-gnu/libQtCore.so.4
#52 0x00007ffff728db2a in QCoreApplication::exec() ()
    at /lib/x86_64-linux-gnu/libQtCore.so.4
#53 0x0000555555688ac4 in main(int, char**) (argc=1, argv=0x7fffffffe1f8)
    at /home/wb/build/lmms_build/test_xz_debuging/lmms/src/core/main.cpp:926

**Comment by firewall1110** [comment]
With stable release (1.2.2) compiled with QT4 i found, that Hung in 

	void waitUntilRead()
	{
		m_writer_sem.acquire( m_size ); /* can not get this line done : but it is QT library call */
		m_writer_sem.release( m_size );
	}
//fifo_buffer.h :: line 69

I made some fprintf(stderr, ...) to trace problem with session (last project DnB) exporting mp3 with default setting.
[----] With sdl device:
---- Mixer::stopProcessing():: 0 
---- m_fifoWriter 
---- m_fifoWriter->finish() 
---- fifoWriter::run() -> write( NULL ) 
m_fifo->available() == True
--FB-- fifoBuffer::waitUntilRead()::m_size == 2 :: have = 2 
--FB-- m_writer_sem.acquire( m_size )
--FB-- m_writer_sem.release( m_size )
---- fifoWriter::run() -> waitUntilRead() 
---- m_fifoWriter->wait() 
---- m_audioDev->stopProcessing() 
---- Mixer::stopProcessing():: 1 
---- m_fifoWriter == NULL
QMetaMethod::invoke: Unable to handle unregistered datatype 'MidiTime'
--RM-- ~RenderManager() 
--M-- restoreAudioDevice() 
---- Mixer::stopProcessing():: 2 
---- m_fifoWriter == NULL
--M-- stopProcessing()
--RM-- mixer()->restoreAudioDevice()
---- Mixer::stopProcessing():: 3 
---- m_fifoWriter 
---- m_fifoWriter->finish() 
---- fifoWriter::run() -> write( NULL ) 
m_fifo->available() == True
--FB-- fifoBuffer::waitUntilRead()::m_size == 2 :: have = 1 
--FB-- m_writer_sem.acquire( m_size )
--FB-- m_writer_sem.release( m_size )
---- fifoWriter::run() -> waitUntilRead() 
---- m_fifoWriter->wait() 
---- m_audioDev->stopProcessing() 
--RM-- mixer()->changeQuality( m_oldQualitySettings )
---- Mixer::stopProcessing():: 4 
---- m_fifoWriter 
---- m_fifoWriter->finish() 
---- fifoWriter::run() -> write( NULL ) 
m_fifo->available() == True
--FB-- fifoBuffer::waitUntilRead()::m_size == 2 :: have = 2 
--FB-- m_writer_sem.acquire( m_size )
--FB-- m_writer_sem.release( m_size )
---- fifoWriter::run() -> waitUntilRead() 
---- m_fifoWriter->wait() 
---- m_audioDev->stopProcessing() 

[----] With jackd device:
---- Mixer::stopProcessing():: 0 
---- m_fifoWriter 
---- m_fifoWriter->finish() 
---- fifoWriter::run() -> write( NULL ) 
m_fifo->available() == False
--FB-- fifoBuffer::waitUntilRead()::m_size == 2 :: have = 0 
--FB-- m_writer_sem.acquire( m_size )
--FB-- m_writer_sem.release( m_size )
---- fifoWriter::run() -> waitUntilRead() 
---- m_fifoWriter->wait() 
---- m_audioDev->stopProcessing() 
---- Mixer::stopProcessing():: 1 
---- m_fifoWriter == NULL
QMetaMethod::invoke: Unable to handle unregistered datatype 'MidiTime'
--RM-- ~RenderManager() 
--M-- restoreAudioDevice() 
---- Mixer::stopProcessing():: 2 
---- m_fifoWriter == NULL
--M-- stopProcessing()
--RM-- mixer()->restoreAudioDevice()
---- Mixer::stopProcessing():: 3 
---- m_fifoWriter 
---- m_fifoWriter->finish() 
---- fifoWriter::run() -> write( NULL ) 
m_fifo->available() == True
--FB-- fifoBuffer::waitUntilRead()::m_size == 2 :: have = 1 

[Actual files are:  fifo_buffer.h   ,    Mixer.cpp  ,   RenderManager.cpp ]

**Comment by firewall1110** [comment]
I found more bugs in rendering:

the same is in **portaudio** device:
[using **direct sound card name**:]
[lmms_portaudio.txt](https://github.com/LMMS/lmms/files/5244438/lmms_portaudio.txt)
[using **oss** "virtual" device:]
[lmms_portaudio_oss.txt](https://github.com/LMMS/lmms/files/5244454/lmms_portaudio_oss.txt)

But with **soundio** device there is **SegFault on export start** (exception with compiled version, but infinite loop with AppImage 1.2.2):
[lmms_soundio_dummy.txt](https://github.com/LMMS/lmms/files/5244475/lmms_soundio_dummy.txt)
[lmms_soundio_pulse.txt](https://github.com/LMMS/lmms/files/5244479/lmms_soundio_pulse.txt)

P.S.
All bugs are both in stable and master versions (in lmms-1.2.2-linux-x86_64.AppImage I can not set direct sound card with portaudio - no sound).

**Comment by firewall1110** [comment]
I clear this:
[lmms_jackd_freeze_gdb.txt](https://github.com/LMMS/lmms/files/5244522/lmms_jackd_freeze_gdb.txt)

But I think it is not more actual ...

**Comment by firewall1110** [comment]
[To PhysSong]
Yes , it is the same bug. 

This can be formulated:

**_Some (playable) devices have problems in Mixer::stopProcessing():
soundio : Segmentation Fault

jackd and (sometimes) portaudio : 
     no sound after  Mixer::stopProcessing() , Mixer::startProcessing() ; 
     and hung in  next Mixer::stopProcessing() ._** 

P.S.

_In my experiment I replaced_ 
    void Mixer::restoreAudioDevice() 
_and_ 
     void Mixer::changeQuality( const struct qualitySettings & _qs )

_with_

void Mixer::restoreAudioDeviceWithQuality( const struct qualitySettings & _qs )

_In original code_
Mixer::stopProcessing() //<- befor rendering (export)
and
Mixer::startProcessing() ;  Mixer::stopProcessing() ; Mixer::startProcessing() ;
after rendering (export)

**Comment by firewall1110** [comment]
**Sorry for my mistake ...**

**soundio crash with segmentation Fault in Mixer::startProcessing() ;** 

but problem is in Mixer::stopProcessing()

Line in AudioSoundIo.cpp:
void AudioSoundIo::stopProcessing()
{
	m_stopped = true;
	if (m_outstream)
	{
		soundio_outstream_destroy(m_outstream); 
// line 228 :: after this we can not simply soundio_outstream_start(m_outstream)) - Segmantation Fault!
		m_outstream = NULL; // Segmantation Fault in every conditions
	}

	if (m_outBuf)
	{
		delete[] m_outBuf;
		m_outBuf = NULL;
	}
}

**Comment by firewall1110** [comment]
It seems that this bug can be fixed by adding only one line:

```
void AudioJack::startProcessing()
{
	if( m_active || m_client == NULL )
	{
		m_stopped = false; // Only this should be added
		return;
	}
...
}
```

Patch file:

[lmms_5664.diff.zip](https://github.com/LMMS/lmms/files/5252516/lmms_5664.diff.zip)

But I have one pending PR ...

**Comment by qnebra** [comment]
It doesn't matter how many pull requests you had open. Made new PR if it fixes issue.

**Comment by firewall1110** [comment]
It seems that I instead have added change to my pending PR  ...
Is it problem?

**Comment by PhysSong** [comment]
Not that much at this moment. We can split them before merging if we should.

**Comment by firewall1110** [comment]
I think it **would be better to work with them both** (_not_ splitting): second change is too small - only 1 line.

**Comment by firewall1110** [comment]
PR patch against stable LMMS 1.2.2 :
[lmms_1_2_2_stable.patch.gz](https://github.com/LMMS/lmms/files/5279245/lmms_1_2_2_stable.patch.gz)

**Comment by DomClark** [comment]
Fixed in #5681.

**Comment by trebmuh** [comment]
Awesome.
Thank you all.
